pkg/apis/projectcalico/v3/nodestatus.go

```diff
@@ -0,0 +1,201 @@
+	V6Peers []CalicoNodePeer `json:"v6Peers,omitempty"`
+}
+
+// CalicoNodeRouteStatus defines the observed state of routes status on the node.
+type CalicoNodeRouteStatus struct {
+	// V4 represents IPv4 routes on the node.
+	V4 []CalicoNodeRoute `json:"bird4,omitempty"`
+
+	// V6 represents IPv6 routes on the node.
+	V6 []CalicoNodeRoute `json:"bird6,omitempty"`
+}
+
+// CalicoNodeBirdStatus defines the observed state of bird.
+type CalicoNodeBirdStatus struct {
+	// Ready indicates if bird status is ready.
+	Ready bool `json:"ready,omitempty"`
+
+	// Bird version.
+	Version string `json:"version,omitempty"`
+
+	// Route ID used by bird.
+	RouteID string `json:"routeID,omitempty"`
+
+	// ServerTime holds the value of serverTime from birdctl output.
+	ServerTime string `json:"serverTime,omitempty"`
+
+	// LastBootTime holds the value of lastBootTime from birdctl output.
+	LastBootTime string `json:"lastBootTime,omitempty"`
+
+	// LastReconfigTime holds the value of lastReconfigTime from birdctl output.
+	LastReconfigTime string `json:"lastReconfigTime,omitempty"`
+}
+
+// CalicoNodePeer contains the status of BGP peers on the node.
+type CalicoNodePeer struct {
+	// IP address of the peer whose condition we are reporting.
+	// If port number is given, format should be `[<IPv6>]:port` or `<IPv4>:<port>` for IPv4.
+	// If optional port number is not set, and this peer IP and ASNumber belongs to a calico/node
+	// with ListenPort set in BGPConfiguration, then we use that port to peer.
+	PeerIP string `json:"peerIP,omitempty"`
+
+	// The type is type of bgp session state.
+	Type BGPPeerType `json:"type,omitempty"`
+
+	// The state is the bgp session state.
+	State string `json:"state,omitempty"`
+
+	// Since is the time since the condition last changed.
+	Since string `json:"since,omitempty"`
+
+	// The reason it's in the current state.
+	Reason string `json:"reason,omitempty"`
+}
+
+// CalicoNodeRoute contains the status of BGP routes on the node.
+type CalicoNodeRoute struct {
+	// Destination of the route.
+	Destination string `json:"destination,omitempty"`
+
+	// Gateway for the destination.
+	Gateway string `json:"gateway,omitempty"`
+
+	// Interface for the destination
+	Interface string `json:"interface,omitempty"`
+
+	// LearnedFrom indicates who installed this route.
+	// If it is populated by a BGP peer, this is the name of the BGPPeer object.
+	// If it is populated by node mesh, this is the name of the node.
+	// Or it is one of kernel, direct or static.
+	LearnedFrom string `json:"learnedFrom,omitempty"`
+}
+
+// NewCalicoNodeStatus creates a new (zeroed) CalicoNodeStatus struct with the TypeMetadata initialised to the current
+// version.
+func NewCalicoNodeStatus() *CalicoNodeStatus {
+	return &CalicoNodeStatus{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       KindCalicoNodeStatus,
+			APIVersion: GroupVersionCurrent,
+		},
+	}
+}
+
+type NodeStatusClassType string
+
+const (
+	NodeStatusClassTypeAgent NodeStatusClassType = "Agent"
```

Comment:
Hmm, I don't think "Agent" makes sense, unless it is assumed that CalicoNodeStatus is only for routing-related information.  Or unless it is expected that more fields will be added to CalicoNodeAgentStatus to report the liveness of Felix, confd etc.

Reply:
Naming is hard. Hmm.. I would expect information from Felix, confd etc. will be added later.